f3.go

```diff
@@ -208,25 +211,34 @@ func (m *F3) Start(startCtx context.Context) (_err error) {
 			select {
 			case update := <-m.manifestProvider.ManifestUpdates():
 				metrics.manifestsReceived.Add(m.runningCtx, 1)
-				if pendingManifest != nil && !manifestChangeTimer.Stop() {
+				if hasPendingManifest && !manifestChangeTimer.Stop() {
 					<-manifestChangeTimer.C
 				}
-				pendingManifest = update
+				if err := m.stopInternal(m.runningCtx); err != nil {
+					// Don't fail here, just log and move on.
+					log.Errorw("failed to stop running F3 instance", "error", err)
+				}
+				metrics.manifestsReceived.Add(m.runningCtx, 1)
+				m.manifest.Store(update)
 			case <-manifestChangeTimer.C:
 			case <-m.runningCtx.Done():
 				return nil
 			}
 
-			if delay, err := m.computeBootstrapDelay(pendingManifest); err != nil {
+			if delay, err := m.computeBootstrapDelay(); err != nil {
 				return err
 			} else if delay > 0 {
+				hasPendingManifest = true
 				manifestChangeTimer.Reset(delay)
 				log.Infow("waiting for bootstrap epoch", "duration", delay.String())
 			} else {
-				if err := m.reconfigure(m.runningCtx, pendingManifest); err != nil {
-					return fmt.Errorf("failed to reconfigure F3: %w", err)
+				if err := m.startInternal(m.runningCtx); err != nil {
+					// There's not much we can do here but complain and wait for
+					// a new manifest. This matches the behavior when we
+					// initially start as well.
+					log.Errorw("failed to start F3", "error", err)
 				}
-				pendingManifest = nil
+				hasPendingManifest = false
```

Comment:
pending set to false even if start internal fails. That's fine and the expectation is that we will retry eventually. Right?

Reply:
The expectation is that retrying wouldn't help. We'll either get a new manifest over the network, or we'll stay in the failed state until we restart.